### PR TITLE
[SPARK-38639] Support cleanedSourcePrint flag to print Codegen cleanedSource

### PR DESCRIPTION
**Pull request by TonyDoen**
### What changes were proposed in this pull request?
adding a config: spark.sql.codegen.cleanedSourcePrint triggers to print codegen clean source

### Why are the changes needed?
When we use spark-sql, encountering problems in codegen source, we often have to change the log level to DEBUG, but there are too many logs in this mode (DEBUG) .
Then `spark.sql.codegen.cleanedSourcePrint` can ensure that just printing codegen source.

### Does this PR introduce _any_ user-facing change?
Yes, add new config: `spark.sql.codegen.cleanedSourcePrint`

### How was this patch tested?
Existed UT

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -1472,6 +1472,14 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val WHOLESTAGE_CODEGEN_CLEAN_SOURCE_ENABLED = buildConf("spark.sql.codegen.cleanedSourcePrint")
+      .internal()
+      .doc("When true, and `spark.sql.codegen.wholeStage` is true, the whole stage " +
+        "(of multiple operators) will be compiled into single java method that will be printed")
+      .version("3.2.0")
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:1479]: Should be 3.4.0.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala [hunk 2]
```diff
@@ -709,7 +709,12 @@ case class WholeStageCodegenExec(child: SparkPlan)(val codegenStageId: Int)
     val duration = System.nanoTime() - startTime
     WholeStageCodegenExec.increaseCodeGenTime(duration)
 
-    logDebug(s"\n${CodeFormatter.format(cleanedSource)}")
+    val formatCleanCode = CodeFormatter.format(cleanedSource)
+    if (conf.wholeStageCleanCodePrintEnabled) {
+      logInfo(s"\n$formatCleanCode")
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala:714]: I don't think it's worth introducing another PR. I believe you can already configure this via log4j configurations

Comment (TonyDoen) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala:714]: Thanks for reviewing. If I turn log level to DEBUG, there are logs more than I need. So I try to configure this on the other way

Comment (LuciferYang) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala:714]: +1 Agree with @HyukjinKwon 

@TonyDoen  you can just configure `org.apache.spark.sql.execution.WholeStageCodegenExec` to debug

**Pull request closed without merging**